ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -353,9 +353,16 @@ public void maybeSetText(ReactTextUpdate reactTextUpdate) {
         new SpannableStringBuilder(reactTextUpdate.getText());
     manageSpans(spannableStringBuilder);
     mContainsImages = reactTextUpdate.containsImages();
+    boolean sameText = sameTextForSpan(getText(), spannableStringBuilder, 0, length());
     mIsSettingTextFromJS = true;
     getText().replace(0, length(), spannableStringBuilder);
     mIsSettingTextFromJS = false;
+    if (!sameText) {
+      InputMethodManager imm = (InputMethodManager) getContext()
+          .getSystemService(Context.INPUT_METHOD_SERVICE);
+      if (imm != null)
```

Comment:
Never use if statements without wrapped brackets, even one line if statements. You're just asking for your future self to be delivered a world of pain.